autometa/config/databases.py

```diff
@@ -66,230 +97,504 @@ class Databases:
+            return False
+        else:
+            return missing
+
+    def download_missing(self, section=None):
+        """Download missing Autometa database dependencies from provided `section`.
+        If no `section` is provided will check all sections.
+
+        Parameters
+        ----------
+        section : str, optional
+            Section to check for missing database files (the default is None).
+            Choices include 'ncbi' and 'markers'.
+
+        Returns
+        -------
+        NoneType
+            Will update provided `section` in `self.config`.
+
+        Raises
+        -------
+        ValueError
+            Provided `section` does not match 'ncbi' and 'markers'.
+
+        """
+        dispatcher = {'ncbi':self.download_ncbi_files, 'markers':self.download_markers}
+        if section and section not in dispatcher:
+            raise ValueError(f'{section} does not match "ncbi" or "markers"')
+        if section:
+            missing = self.get_missing(section=section)
+            options = missing.get(section, [])
+            dispatcher[section](options)
+        else:
+            self.get_missing()
+            for section,options in missing.items():
+                dispatcher[section](options)
 
-    def update_missing(self):
-        """Download and format databases for all options in each section.
+    def compare_checksums(self, section=None, validate=False):
+        """Get all invalid database files in `options` from `section`
+        in config. An md5 checksum comparison will be performed between the
+        current and file's remote md5 to ensure file integrity prior to checking
+        the respective file as valid.
 
-        NOTE: This will only perform the download and formatting if self.dryrun is False
+        Parameters
+        ----------
+        section : str, optional
+            Configure provided `section`. Choices include 'markers' and 'ncbi'.
+            (default will download/format all database directories)
+        validate : bool, optional
+            check whether database files are invalid and exit (default is False).
+
+        Returns
+        -------
+        bool or dict
+            * if `validate` is True : bool
+                False if all database files are valid otherwise True.
+            * if `validate` is False : dict
+                {section:{option, option,...}, section:{...}, ...}
+
+        """
+        sections = [section] if section else Databases.SECTIONS.keys()
+        invalid = {}
+        taxdump_checked = False
+        for section in sections:
+            for option in self.config.options(section):
+                if option not in Databases.SECTIONS.get(section):
+                    # Skip user added options not required by Autometa
+                    continue
+                # nodes.dmp, names.dmp and merged.dmp are all in taxdump.tar.gz
+                option = 'taxdump' if option in {'nodes','names','merged'} else option
```

Comment:
I made another comment elsewhere but here we are just checking that the parent taxdump file is OK but not the untarred component files.